HikVisionSource/README.md

```diff
@@ -0,0 +1,257 @@
+# Overview
+
+The following documentation outlines how to incorporate a HikVision  Source as part of your project. This allows a user to construct applications that detect creation of hikVision files in pre defined folder and upload the file content as events to vantiq. the extension source enable control on the names of the idfferent attribute base on the order in the hikVision file and change the name or delete after procsiing the file. 
+The extension source can handle multiple files parralel , this can be controled by parameters in the config section . 
+
+In order to incorporate this Extension Source, you will need to create the Source in the VANTIQ Modelo IDE. The documentation has been split into two parts, [Setting Up Your Machine](#machine) and [Setting Up Your VANTIQ Modelo IDE](#vantiq).
+
+# Prerequisites <a name="pre" id="pre"></a>
+
+**IMPORTANT:** Read the [Testing](#testing) section before building this project.
+
+An understanding of the VANTIQ Extension Source SDK is assumed. Please read the [Extension Source README.md](../README.md) for more 
+information.
+
+The user must [define the HikVision Source implementation](../README.md#-defining-a-typeimplementation) in the VANTIQ Modelo IDE. For an example of the definition, 
+please see the [*hikVisionImpl.json*](src/test/resources/hikVisionImpl.json) file located in the *src/test/resources* directory.
+
+Additionally, an example project named *HikVisionExample.zip* can be found in the *src/test/resources* directory.
+
+# Setting Up Your Machine <a name="machine" id="machine"></a>
+
+## Repository Contents
+
+*   **HikVisionMain** -- The main function for the program. Connects to sources as specified in a
+    configuration file.
+*   **HikVisionCore** -- Coordinates the connections to Vantiq , responsible for recovery the connection with Vantiq Server
+*   **HikVisionHandleConfiguration** -- Sets up the trigger to the file system for detect and processed new hikVision file 
+*   **HikVision** -- The class that directly interacts with the file system watch service , detect the file and process it . 
+
+*  **HCNetSDK** -- The implemetation of the hikVoision jna interface. 
+
+
+*  **NOTE** -- for windows machine : additional `windows.zip` contains HikVision window based device implementation must be installed , in order to get the Camera communication , please refer to `How to run for distribution` section
+
+*  **NOTE** -- for Unix based system , please use `HikVisionSource_wLinux.tgz`
```

Comment:
We'll want to include lines of where the customers should get these, where they should put them, so that we can package the application appropriately.